### PR TITLE
Add test cases for non-breaking whitespace characters for Codes

### PR DESCRIPTION
**Pull request by MichaelSwigerAtBentley**
Expands Code tests to include non-breaking whitespace character (`\xa0`) edge cases, for transformer

### CODE REVIEW

File: core/backend/src/test/imodel/Code.test.ts [hunk 1]
```diff
@@ -23,292 +23,409 @@ describe("Code insertion tests", () => {
...
+    it("should get undefined when querying an element with an empty code", () => {
+      assert.exists(imodel.elements);
+      const emptyCode = Code.createEmpty();
+      const elementId = imodel.elements.queryElementIdByCode(emptyCode);
+      assert.isUndefined(elementId);
+    });
+
+    it("should get undefined when querying an element with a non-breaking space value", () => {
+      assert.exists(imodel.elements);
+      const nonBreakingSpaceCode = new Code({ scope: "0x11", spec: "0x10", value: "\xa0" });
```
Comment (DanRod1999) [inline core/backend/src/test/imodel/Code.test.ts:64]: does an element in the imodel exist with this code value? Since we don't accept this space values I assume the code in reality would be {... value: ""}. Not sure if that is a legal value in general though.

Is this just checking that it returns undefined and not an error?

File: core/backend/src/test/imodel/Code.test.ts [hunk 2]
```diff
@@ -23,292 +23,409 @@ describe("Code insertion tests", () => {
...
+      assert.equal(updatedElement3.code.value, newCode.value);
+      assert.equal(updatedElement3.code.spec, newCode.spec);
+      assert.equal(updatedElement3.code.scope, newCode.scope);
+
+      newCode = new Code({ scope: "0x1", spec: "0x11", value: "\xa0" }); // non-breaking space value
+      element.code = newCode;
+      element.update();
+
+      const updatedElement4 = imodel.elements.getElement(elementId);
+      assert.equal(updatedElement4.code.value, "");
```
Comment (DanRod1999) [inline core/backend/src/test/imodel/Code.test.ts:417]: do we have any docs or comments to explain why we expect this behavior. For example is this just for trailing spaces, or would something like "hello\xa0world" turn into "helloworld"?

File: core/backend/src/test/imodel/Code.test.ts (hunk 1 above)
Comment (MichaelSwigerAtBentley) [reply core/backend/src/test/imodel/Code.test.ts:64]: An element with "" is inserted, but not \xa0 (which just inserts as "" anyway)

I could add an insert though to these tests to make it explicit

File: core/backend/src/test/imodel/Code.test.ts (hunk 2 above)
Comment (MichaelSwigerAtBentley) [reply core/backend/src/test/imodel/Code.test.ts:417]: I don't think so, https://www.itwinjs.org/bis/guide/fundamentals/codes/ has some notes about value uniqueness, but does not go into specifics. I think it should be outlined there

It only trims trailing spaces, for instance, code values with multiple words like "this is my code value" are fine

File: core/backend/src/test/imodel/Code.test.ts (hunk 1 above)
Comment (MichaelSwigerAtBentley) [reply core/backend/src/test/imodel/Code.test.ts:64]: Added
